### PR TITLE
executor, plan: add revoke role support

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Support revoke roles from users.

### What is changed and how it works?
Add RevokeRoleStmt structure which could parser REVOKE FROM SQL. Executor in TiDB will read information from RevokeRoleStmt to delete relation from mysql.role_edges.

Executor will begin transaction to delete every information in `mysql.role_edges` which match delete conditions.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

 - Increased code complexity

### CODE REVIEW

**Comment by jackysp** [comment]
Please remove the test output.

**Comment by tiancaiamao** [comment]
Ping @imtbkcat

**Comment by imtbkcat** [comment]
parser pr: https://github.com/pingcap/parser/pull/245

**Comment by imtbkcat** [comment]
PTAL @jackysp @tiancaiamao

File: go.mod [hunk 1]
```diff
@@ -90,3 +90,5 @@ require (
 	sourcegraph.com/sourcegraph/appdash v0.0.0-20180531100431-4c381bd170b4
 	sourcegraph.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67
 )
+
+replace github.com/pingcap/parser => github.com/imtbkcat/parser v0.0.0-20190401043336-a458e4f8a051
```
Comment (tiancaiamao) [inline go.mod]: Parse has been merged, please update here @imtbkcat

File: executor/simple_test.go [hunk 2]
```diff
@@ -141,6 +141,21 @@ func (s *testSuite3) TestRole(c *C) {
...
+	tk.MustExec("insert into mysql.role_edges (FROM_HOST,FROM_USER,TO_HOST,TO_USER) values ('%','r_1','%','root')")
+	tk.MustExec("insert into mysql.role_edges (FROM_HOST,FROM_USER,TO_HOST,TO_USER) values ('%','r_2','%','root')")
+	_, err = tk.Exec("revoke test@localhost, r_1 from root;")
+	c.Check(err, IsNil)
+	_, err = tk.Exec("revoke `r_2`@`%` from root, u_2;")
+	c.Check(err, IsNil)
+	result = tk.MustQuery(`SELECT * FROM mysql.default_roles WHERE DEFAULT_ROLE_USER="test" and DEFAULT_ROLE_HOST="localhost"`)
+	result.Check(nil)
+	dropRoleSQL = `DROP ROLE 'test'@'localhost', r_1, r_2;`
+	tk.MustExec(dropRoleSQL)
```
Comment (tiancaiamao) [inline executor/simple_test.go]: Cover a negative case?

File: go.mod (hunk 1 above)
Comment (imtbkcat) [reply go.mod]: ok

File: executor/simple_test.go (hunk 2 above)
Comment (imtbkcat) [reply executor/simple_test.go]: @tiancaiamao added.

**Comment by imtbkcat** [comment]
PTAL @tiancaiamao

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**